runtime/fundamentals/testing.md

```diff
@@ -0,0 +1,757 @@
+are up-to-date and functional.
+
+### Example code blocks
+
+````ts title="example.ts"
+/**
+ * # Examples
+ *
+ * ```ts
+ * const x = 42;
+ * ```
+ */
+````
+
+The triple backticks mark the start and end of code blocks, the language is
+determined by the language identifier attribute which may be `js`, `jsx`, `ts`
+or `tsx`. If no language identifier is specified then the language is inferred
+from media type of the source document that the code block is extracted from.
+
+```sh
+deno test --doc example.ts
+```
+
+The above command will extract this example, and then type-check it as a
+standalone module living in the same directory as the module being documented.
+
+### Documenting exports
+
+To document your exports, import the module using a relative path specifier:
+
+````ts
+/**
+ * # Examples
+ *
+ * ```ts
+ * import { foo } from "./foo.ts";
+ * ```
+ */
+export function foo(): string {
+  return "foo";
+}
+````
+
+## Sanitizers
+
+The test runner offers several sanitizers to ensure that the test behaves in a
+reasonable and expected way.
+
+### Resource sanitizer
+
+Ensures that all resources created during a test are closed to prevent leaks.
+Enabled by default, it can be disabled with `sanitizeResources: false`:
+
+```ts
+Deno.test({
+  name: "leaky resource test",
+  async fn() {
+    await Deno.open("hello.txt");
+  },
+  sanitizeResources: false,
+});
+```
+
+### Op sanitizer
+
+Ensures that all async operations started in a test are completed before the
+test ends. Enabled by default, it can be disabled with `sanitizeOps: false`:
+
+```ts
+Deno.test({
+  name: "leaky operation test",
+  fn() {
+    crypto.subtle.digest(
+      "SHA-256",
+      new TextEncoder().encode("a".repeat(100000000)),
+    );
+  },
+  sanitizeOps: false,
+});
+```
+
+### Exit sanitizer
+
+Ensures that tested code doesn’t call `Deno.exit()`, which could signal a false
+test success. Enabled by default, it can be disabled with `sanitizeExit: false`.
+
+```ts
+Deno.test({
+  name: "false success",
+  fn() {
+    Deno.exit(0);
+  },
+  sanitizeExit: false,
+});
+
+// This test never runs, because the process exits during "false success" test
+Deno.test({
+  name: "failing test",
+  fn() {
+    throw new Error("this test fails");
+  },
+});
+```
+
+## Snapshot testing
+
+The [Deno standard library](/runtime/fundamentals/standard_library/) includes a
+[snapshot module](https://jsr.io/@std/testing/doc/snapshot/~) that allows
+developers to write tests by comparing values against reference snapshots. These
+snapshots are serialized representations of the original values and are stored
+alongside the test files.
+
+Snapshot testing enables catching a wide array of bugs with very little code. It
+is particularly helpful in situations where it is difficult to precisely express
+what should be asserted, without requiring a prohibitive amount of code, or
+where the assertions a test makes are expected to change often.
+
+---
+
```

Comment:
Everything below here is about the LSP protocol integration. I feel like this should be a separate page about "integrating with the LSP" or something. It's not relevant for folks wanting to write tests with Deno. The only nice tip is that you can pass additional arguments to tests started from one of the code lenses by adding it in VSCode's settings file.

Reply:
I've moved it to its own page for now in tools, which is where the CLI reference is, when we move that all over to the new information architecture we can reconsider where it might be best to go.